hooks/gen_docs.py

```diff
@@ -1,10 +1,277 @@
+    os.listdir(PATH_DOCS_RESOURCES / "pipeline-defaults/headers")
+)
+
+
+class KpopsComponent(NamedTuple):
+    # name: str
+    attrs: list[str]
+    specific_attrs: list[str]
+
+
+def filter_sections(
+    component_name: str, sections: list[str], include_inherited: bool = False
+) -> list[str]:
+    """Find all sections that are specific to a component from a list
+
+    :param component_name: Component name
+    :param sections: Available section files, names with extension, no path
+    :param include_inherited: Whether to include sections that can be defined in
+        a parent component defaults
+    :returns: The applicable sections
+    """
+    component_sections: list[str] = []
+    for target_section in KPOPS_COMPONENTS_FIELDS[component_name]:
+        component = component_name
+        section = target_section + "-" + component + ".yaml"
+        if section in sections:
+            component_sections.append(section)
+        elif (
+            not include_inherited
+            and KPOPS_COMPONENTS_INHERITANCE_REF[component]
+            == PipelineComponent.get_component_type()
+        ):
+            section = target_section + ".yaml"
+            if section in sections:
+                component_sections.append(section)
+        elif include_inherited:
+            while component != PipelineComponent.get_component_type():
+                component = KPOPS_COMPONENTS_INHERITANCE_REF[component]
+                section = target_section + "-" + component + ".yaml"
+                if section in sections:
+                    component_sections.append(section)
+                    break
+                elif component == PipelineComponent.get_component_type():
+                    section = target_section + ".yaml"
+                    if section in sections:
+                        component_sections.append(section)
+                        break
+    component_sections.sort(key=component_section_position_in_definition)
+    return component_sections
+
+
+def concatenate_text_files(*sources: Path, target: Path) -> None:
+    """Concatenate the given files into one
+
+    :param *sources: Files to be conatenated. The order of the inputs will be
+        retained in the result
+    :param target: File to store the result in. Note that any previous contents
+        will be deleted.
+    """
+    with target.open("w+") as f:
+        for source in sources:
+            f.write(source.read_text())
+
+
+def component_section_position_in_definition(key: str) -> int:
+    """Returns a positional value for a given str if found in the list of
+    component definition sections
+
+    :param key: A value to look for in the list of sections
+    :returns: The corresponding position or 0 if not found
+    """
+    for section in SECTIONS_ORDER:
+        if key.startswith((section + "-", section + ".")):
+            return SECTIONS_ORDER.index(section)
+    return 0
+
+
+def check_for_changes_in_kpops_component_structure() -> bool:
```

Comment:
do we still need this? Now that the pre-commit hook is triggered by specific files changing, we could just assume it's necessary to re-generate the docs, no?

Reply:
This decides whether to regenerate the dependencies. If it returns `False`, the docs are generated using the dependency yamls. On `True` they are regenerated first. The intention is to save some time and resources.